File: test-agent/todo-cli/storage.py
import json
import os

DATA_FILE = "todos.json"


def load_todos():
    if not os.path.exists(DATA_FILE):
        return []
    with open(DATA_FILE, "r", encoding="utf-8") as f:
        return json.load(f)


def save_todos(todos):
    with open(DATA_FILE, "w", encoding="utf-8") as f:
        json.dump(todos, f, indent=2, ensure_ascii=False)

File: test-agent/todo-cli/storage.js
const fs = require("fs");
const path = require("path");

const DATA_FILE = path.join(__dirname, "todos.json");

function loadTodos() {
  if (!fs.existsSync(DATA_FILE)) return [];
  const raw = fs.readFileSync(DATA_FILE, "utf-8");
  return JSON.parse(raw);
}

function saveTodos(todos) {
  fs.writeFileSync(DATA_FILE, JSON.stringify(todos, null, 2), "utf-8");
}

module.exports = { loadTodos, saveTodos };
